### PR TITLE
Remove cffi dependency as it doesn't look like we're using it

### PR DESCRIPTION
**Pull request by albanD**
Maybe this will go horribly wrong in CI but works fine without it locally!

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Review by kit1980 — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -8,7 +8,6 @@ requires = [
     "pyyaml",
     "setuptools",
     "cmake",
-    "cffi",
     "typing_extensions",
     "future",
```
Comment (Skylion007) [inline pyproject.toml:12]: Huh, didn't we just kill future and six? Seems like we missed a spot

**Review by seemethere — APPROVED** [review]

File: .circleci/docker/common/install_conda.sh [hunk 2]
```diff
@@ -75,7 +75,7 @@ if [ -n "$ANACONDA_PYTHON_VERSION" ]; then
   }
 
   # Install PyTorch conda deps, as per https://github.com/pytorch/pytorch README
-  CONDA_COMMON_DEPS="astunparse pyyaml mkl=2022.0.1 mkl-include=2022.0.1 setuptools cffi six"
+  CONDA_COMMON_DEPS="astunparse pyyaml mkl=2022.0.1 mkl-include=2022.0.1 setuptools six"
```
Comment (vadimkantorov) [inline .circleci/docker/common/install_conda.sh:78]: Actually, is six still needed? I think it was there for python2

Comment (albanD) [reply .circleci/docker/common/install_conda.sh:78]: I don't think it is strictly needed no as we don't need these constructs anymore. But it is used quite a bit in the codebase. So we would need to remove use of it first before being able to remove it from the builds.

Comment (vadimkantorov) [reply .circleci/docker/common/install_conda.sh:78]: will create an issue for dropping it eventually

**Pull request closed without merging**